Reporter unit sometimes changes order and breaks

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

Observed in the first run of https://github.com/Kong/kubernetes-ingress-controller/actions/runs/2098352423, cleared on re-run:

```
2022-04-05T19:32:32.1131142Z === RUN   TestReporterOnce
2022-04-05T19:32:32.1131534Z     reports_test.go:109: 
2022-04-05T19:32:32.1131991Z         	Error Trace:	reports_test.go:109
2022-04-05T19:32:32.1132365Z         	Error:      	Not equal: 
2022-04-05T19:32:32.1134244Z         	            	expected: "v=kic.version;k8sv=k8s.version;kv=kong.version;db=off;id=6acb7447-eedf-4815-a193-d714c5108f7b;hn=example.local;feature-knative=false;feature-gateway=true;"
2022-04-05T19:32:32.1136284Z         	            	actual  : "v=kic.version;k8sv=k8s.version;kv=kong.version;db=off;id=6acb7447-eedf-4815-a193-d714c5108f7b;hn=example.local;feature-gateway=true;feature-knative=false;"
2022-04-05T19:32:32.1136846Z         	            	
2022-04-05T19:32:32.1137218Z         	            	Diff:
2022-04-05T19:32:32.1137714Z         	            	--- Expected
2022-04-05T19:32:32.1138132Z         	            	+++ Actual
2022-04-05T19:32:32.1138688Z         	            	@@ -1 +1 @@
2022-04-05T19:32:32.1140423Z         	            	-v=kic.version;k8sv=k8s.version;kv=kong.version;db=off;id=6acb7447-eedf-4815-a193-d714c5108f7b;hn=example.local;feature-knative=false;feature-gateway=true;
2022-04-05T19:32:32.1143046Z         	            	+v=kic.version;k8sv=k8s.version;kv=kong.version;db=off;id=6acb7447-eedf-4815-a193-d714c5108f7b;hn=example.local;feature-gateway=true;feature-knative=false;
2022-04-05T19:32:32.1143830Z         	Test:       	TestReporterOnce
2022-04-05T19:32:32.1144293Z --- FAIL: TestReporterOnce (0.00s)
```

### Expected Behavior

Consistent unit

### Steps To Reproduce

_No response_

### Kong Ingress Controller version

_No response_

### Kubernetes version

_No response_

### Anything else?

_No response_

--- a/internal/util/reports_test.go
+++ b/internal/util/reports_test.go
@@ -8,6 +8,7 @@ import (
 	"net"
 	"os"
 	"strconv"
+	"strings"
 	"sync"
 	"testing"
 
@@ -104,9 +105,27 @@ func TestReporterOnce(t *testing.T) {
 	go runTestTLSServer(ctx, t, listener, reqs)
 
 	reporter.once()
-	want := "v=kic.version;k8sv=k8s.version;kv=kong.version;db=off;" +
-		"id=6acb7447-eedf-4815-a193-d714c5108f7b;hn=example.local;feature-knative=false;feature-gateway=true;"
-	assert.Equal(want, reporter.serializedInfo)
+	got := make(map[string]string)
+	for _, line := range strings.Split(reporter.serializedInfo, ";") {
+		if line == "" {
+			continue
+		}
+		parts := strings.Split(line, "=")
+		k, v := parts[0], parts[1]
+		got[k] = v
+	}
+
+	want := map[string]string{
+		"v":               "kic.version",
+		"k8sv":            "k8s.version",
+		"kv":              "kong.version",
+		"db":              "off",
+		"id":              "6acb7447-eedf-4815-a193-d714c5108f7b",
+		"hn":              "example.local",
+		"feature-knative": "false",
+		"feature-gateway": "true",
+	}
+	assert.Equal(want, got)
 }
 
 func TestReporterSendStart(t *testing.T) {

